python/triton/runtime/jit.py

```diff
@@ -478,7 +479,7 @@ def __init__(self, fn, version=None, do_not_specialize=None, debug=None, noinlin
 
         # function source code (without decorators)
         self.src = textwrap.dedent(inspect.getsource(fn))
-        self.src = self.src[self.src.find("def"):]
+        self.src = self.src[re.search(r"def (\w+)\s*\((.*?)\):", self.src).start():]
```

Comment:
Can we remove the parens around `(.*?)`?

Also if we're not in multiline mode I think this does not work, because `.` only matches on the current line?

Reply:
yeah you're right, sorry I forgot the multiline case.